core/src/main/scala-3/caliban/schema/macros/Macros.scala

```diff
@@ -88,4 +89,26 @@ private[caliban] object Macros {
     Expr(TypeRepr.of[P].typeSymbol.flags.is(Flags.Enum) && TypeRepr.of[T].typeSymbol.flags.is(Flags.Enum))
   }
 
+  def hasOneOfInputAnnotationImpl[T: Type](using q: Quotes): Expr[Boolean] = {
+    import q.reflect.*
+    Expr(TypeRepr.of[T].typeSymbol.annotations.exists(_.tpe.typeSymbol.name == "GQLOneOfInput"))
+  }
+
+  def isValidOneOffInputImpl[T: Type](using q: Quotes): Expr[Boolean] = {
```

Comment:
I think I would rather have it in the schema validation. Your point is right about UX, I am just wary of macros because they might break (like of them broke with Java 17) and aren't easy to maintain.

Schema validation happens when you start your server to you find quickly anyway no?

Reply:
I think you're right - it's nice to have these checks during compiling but they do add unnecessary maintenance (and to some degree, compiling time) overhead. Removed